service/src/main/kotlin/fi/espoo/evaka/reports/FuturePreschoolersReport.kt

```diff
@@ -0,0 +1,196 @@
+        clock: EvakaClock,
+        @RequestParam municipal: Boolean
+    ): List<PreschoolGroupsReportRow> {
+        return db.connect { dbc ->
+                dbc.read {
+                    accessControl.requirePermissionFor(
+                        it,
+                        user,
+                        clock,
+                        Action.Global.READ_FUTURE_PRESCHOOLERS
+                    )
+                    it.setStatementTimeout(REPORT_STATEMENT_TIMEOUT)
+                    it.getPreschoolGroupsRows(municipal)
+                }
+            }
+            .also { Audit.FuturePreschoolers.log(meta = mapOf("count" to it.size)) }
+    }
+}
+
+const val preschoolSelectionAge = 5
+
+fun Database.Read.getFuturePreschoolerRows(): List<FuturePreschoolersReportRow> =
+    createQuery<DatabaseTable> {
+            sql(
+                """
+SELECT p.id, 
+    p.social_security_number AS childSsn,
+    p.last_name AS childLastName,
+    p.first_name AS childFirstName,
+    p.street_address AS childAddress,
+    p.postal_code AS childPostalCode,
+    upper(p.post_office) AS childPostOffice,
+    d.name AS unitName,
+    d.street_address AS unitAddress,
+    d.postal_code AS unitPostalCode,
+    upper(d.post_office) AS unitPostOffice,
+    NULL as unitArea, -- cannot be obtained from current evaka dataset, care area is not precise enough
+    g1.last_name AS guardian1LastName,
+    g1.first_name AS guardian1FirstName,
+    g1.street_address AS guardian1Address,
+    g1.postal_code AS guardian1PostalCode,
+    upper(g1.post_office) AS guardian1PostOffice, 
+    g1.phone AS guardian1Phone,
+    g1.email AS guardian1Email,
+    g2.last_name AS guardian2LastName, 
+    g2.first_name AS guardian2FirstName, 
+    g2.street_address AS guardian2Address,
+    g2.postal_code AS guardian2PostalCode,
+    upper(g2.post_office) AS guardian2PostOffice, 
+    g2.phone AS guardian2Phone,
+    g2.email AS guardian2Email,
+    CASE WHEN sn.shift_care = 'FULL'::shift_care_type OR sn.shift_care = 'INTERMITTENT'::shift_care_type THEN true ELSE false END AS shiftCare,
+    NULL AS languageEmphasisGroup, -- cannot be obtained from current evaka dataset
+    CASE WHEN sno.name_fi like 'Kaksivuotinen%' THEN true ELSE false END AS twoYearPreschool
+FROM person p
+JOIN placement pl ON pl.child_id = p.id
+JOIN daycare d ON d.id = pl.unit_id
+LEFT JOIN (
+    SELECT id, last_name, first_name, street_address, postal_code, post_office, phone, email
+    FROM person
+) g1 ON g1.id = (SELECT head_of_child FROM fridge_child WHERE child_id = p.id)
+LEFT JOIN (
+    SELECT id, last_name, first_name, street_address, postal_code, post_office, phone, email
+    FROM person
+) g2 ON g2.id = (SELECT person_id FROM fridge_partner WHERE partnership_id =
+    (SELECT partnership_id FROM fridge_partner WHERE person_id = g1.id) AND person_id != g1.id)
+LEFT JOIN service_need sn ON sn.placement_id = p.id and sn.start_date < current_date AND sn.end_date >= current_date
```

Comment:
You should not use `current_date`, but pass the date as a bind parameter. This makes testing easier and allows using mock clock for example.